lib/easybill-rest-client/api_client.rb

```diff
@@ -26,55 +29,58 @@ def request_collection(method, endpoint, params = {})
     end
 
     def request(method, endpoint, params = {})
-      Retryable.retryable(
-        tries: tries,
-        sleep: retry_cool_off_time,
-        on: EasybillRestClient::TooManyRequests
-      ) do
-        response = perform_request(method, endpoint, params)
-        process_response(response)
+      @request_logger = RequestLogger.new(logger: logger, request_id: Time.now.to_f)
```

Comment:
```ruby
require 'benchmark/ips'
require 'securerandom'

Benchmark.ips do |x|
  x.report("time") { Time.now.to_f.to_s }
  x.report("rand") { SecureRandom.hex }
  x.compare!
end
```

```
Warming up --------------------------------------
                time    53.118k i/100ms
                rand    50.010k i/100ms
Calculating -------------------------------------
                time    669.184k (± 3.9%) i/s -      3.346M in   5.008970s
                rand    588.651k (± 5.0%) i/s -      2.951M in   5.026244s

Comparison:
                time:   669183.8 i/s
                rand:   588651.4 i/s - 1.14x slower
```

So unless you're making more than 100,000 requests per second, I think we're good here 😉